### ISSUE TITLE
Element.replaceChild should allow for `old, new` argument ordering if `new` is a new element and `old` is a child of Element

### ISSUE THREAD

**Issue by gijsk** [whatwg/dom#847]
Concretely, given DOM constructed from this HTML:

```html
<div><span id="a">Old</span></div>
```

Running:

```js
document.querySelector("div").replaceChild(document.querySelector("#a"), document.createTextNode("new"));
```

currently throws a DOM Node not found error (cf. step 3 in https://dom.spec.whatwg.org/#concept-node-replace ).

Really, `old, new` argument ordering would IMO make more sense, but completely changing the argument order is probably not realistic.

However, in the case where (a) the second argument has no parent and (b) the first argument **is** a child of the _parent_, instead of throwing an error, we could just Do What the Developer Means -- instead of throwing, which we do now, which is very developer-unfriendly. I'm not even sure if we need the (a) restriction, but certainly that seems like it'd be safer than also doing it if (a) has a different parent.

Of course, perhaps in web-compat practice there are people who depend on the error throwing behaviour? I don't know...

**Comment by domenic** [comment]
I don't think it's developer-unfriendly to throw an error if you get argument order wrong. Magically switching the argument order in some cases but not others is not good API design.

**Comment by annevk** [comment]
@gijsk why not convert such code to `a.replaceWith(b)`?

**Comment by gijsk** [comment]
> I don't think it's developer-unfriendly to throw an error if you get argument order wrong. Magically switching the argument order in some cases but not others is not good API design.

Well, I think the spec'd argument order is wrong and we're fixing API design that is already bad... Really, I'd prefer it if we just changed the argument order outright, but I also understand that's not likely to be a realistic option, so I went with the next best thing.

> @gijsk why not convert such code to `a.replaceWith(b)`?

I guess that works. The MDN page wasn't particularly explicit that this existed, and cross-browser support is not amazing (but of course wouldn't improve for changes we introduce to `replaceChild` now).

**Comment by annevk** [comment]
Yeah, changes to `replaceChild` would take longer. Also, per MDN `replaceWith` only doesn't work in Internet Explorer, which seems good? Definitely for Firefox's frontend that should be fine. 😊

**Comment by annevk** [comment]
Closing this as WONTFIX as I suspect this will trip people up and we have alternatives now.
